pkg/resource/deploy/step_generator.go

```diff
@@ -623,6 +623,16 @@ func (sg *stepGenerator) generateSteps(event RegisterResourceEvent) ([]Step, err
 		contract.Assertf(n == len(randomSeed),
 			"generated fewer (%d) than expected (%d) random bytes", n, len(randomSeed))
 	}
+	var autonaming *plugin.AutonamingOptions
+	var autonamingDeleteBeforeCreate bool
+	if sg.deployment.opts.Autonaming != nil {
+		a, dbc, err := sg.deployment.opts.Autonaming.AutonamingForResource(urn, randomSeed)
+		if err != nil {
+			return nil, fmt.Errorf("failed to determine autonaming setting: %w", err)
+		}
+		autonaming = a
+		autonamingDeleteBeforeCreate = dbc
```

Comment:
I'm wondering if the smart/safe/correct route here is to simply set `goal.DeleteBeforeReplace = dbc`

Reply:
Yes, I think you are right, we can get to use it as

```
		if dbc {
			goal.DeleteBeforeReplace = &dbc
		}
```

That is, if the autonamer says it's true, we set the goal to true, but don't change it otherwise. It's an "or" which is awkward because of nil-ness here.